src/main/java/com/pusher/rest/Pusher.java

```diff
@@ -341,9 +372,14 @@ public Result trigger(final List<String> channels, final String eventName, final
         Prerequisites.noNullMembers("channels", channels);
         Prerequisites.areValidChannels(channels);
         Prerequisites.isValidSocketId(socketId);
-
-        final String body = BODY_SERIALISER.toJson(new TriggerData(channels, eventName, serialise(data), socketId));
-
+        Prerequisites.eitherOneorNoEncryptedChannels(channels);
+        String body;
+        if(Crypto.isEncryptedChannel(channels.get(0))) {
+            EncryptedPayload dataEncrypted = pusherCrypto.encrypt(channels.get(0), data);
+            body = BODY_SERIALISER.toJson(new TriggerData(channels, eventName, serialise(dataEncrypted), socketId));
+        } else {
+            body = BODY_SERIALISER.toJson(new TriggerData(channels, eventName, serialise(data), socketId));
```

Comment:
This should explicitly fail if multiple channels are passed in and some of them are encrypted. See how this is done in the node library: https://github.com/pusher/pusher-http-node/blob/efa7c88446a1db200f435a65a940ff7de1ecfcfe/lib/events.js#L37-L40

Reply:
This is handled by the prerequisite - unless you mean something else?